config/lobby/db/001_create_tables

```diff
@@ -1,94 +1,42 @@
-create table bad_words (
-    word character varying(40) not null
+create table ta_users (
```

Comment:
I'm still pretty sure we shouldn't be modifying this file _at all_ (other than perhaps adding comments).  Ideally, this file represents the current state of the production database (and, if that's not the case, we either have to modify this file to match production or manually modify production to match this file, as @DanVanAtta did with the `password` column type change a few days ago).  This file will never be applied to the production database again, so the changes here will not be picked up.

Any changes from this PR should be completely captured in the `002_*` file below.

Reply:
I really hope we come up with an automated solution for this.
I'm really not happy with the way we handle changes in the database structure.
I dislike the idea of having one "original" File which is modified by subsequent scripts, this makes reverse engineering really hard, because you can't immediately spot how the database structure is in it's current state:
Is this table still being used, or is it removed in one of the scripts?
Is this datatype final, or do we change it somewhere?
Are there any further restrictions on the content of this column or are checks being removed?
Does this column even exist in the final schema?

All of this causes confusion, which is clearly not the goal here.
But I see your points, and agree that having incremental scripts makes the testing more "real" although I can assure you I tested my altering scripts multiple times against the database scheme of our production DB (a quick sql dump of the scheme makes stuff really easy).
While doing so I noticed another problem with having an original script: TripleA doesn't has the ability to IP-Ban people for a long time now, nevertheless the production postgresdb had a banned_ips and a muted_ips table, which were not used when postgres was introduced. (Tha's why I added the DROP command).

I'd suggest leaving the scripts here as-is, and do the upgrading half-manually, or if anyone wants to integrate liquibase into our workflow I can wait and modify this PR to work with it.

Reply:
> I dislike the idea of having one "original" File which is modified by subsequent scripts

Thoughtworks as a pretty good [article](https://martinfowler.com/articles/evodb.html) on the philosophy behind this technique, specifically with a [section](https://martinfowler.com/articles/evodb.html#AllDatabaseChangesAreMigrations) related to the topic of migrations.

> this makes reverse engineering really hard, because you can't immediately spot how the database structure is in it's current state

Not too hard.  You can get a snapshot of the actual schema by just applying the scripts to a pristine database and then dumping the schema of each table.  This [SO answer](https://stackoverflow.com/a/37571254/3900879) provides some inspiration for how to do that from the shell for a MySQL database.  Using the psql equivalents for `SHOW TABLES` (`\dt`) and `DESCRIBE TABLE` (`\d`) should be straightforward.

Reply:
> I dislike the idea of having one "original" File which is modified by subsequent scripts, this makes reverse engineering really hard, because you can't immediately spot how the database structure is in it's current state:

If you modify one file, then you also do not know it's current state. Were the modification runs? How many modifications have been run and which ones were applied? Does the file just track the DB state, if so what would keep it in sync if anything?

> I really hope we come up with an automated solution for this.

Liquibase is pretty much a tool to do this exactly :)
https://github.com/triplea-game/triplea/issues/2356

> Is this table still being used, or is it removed in one of the scripts?

Often a good question, but if you have one file, then how do you go from DB in state A to state B?  If things get very complex with the DB patches, a 'snapshot' concept can be introduced. Basically you take a snapshot of the schema and data, and re-init the DB patches from that location. So at *that* time you would for example recreate the '001' file and then blow away any other patches since they are already included.

> TripleA doesn't has the ability to IP-Ban people for a long time now, nevertheless the production postgresdb had a banned_ips and a muted_ips table

Wait, TripleA does.. As a moderator in lobby you can right click to ban by IP, right? It's not the most useful of features in its current form, but it is there to my knowledge. 

> I'd suggest leaving the scripts here as-is, and do the upgrading half-manually, or if anyone wants to integrate liquibase into our workflow I can wait and modify this PR to work with it

I would disagree here. I do not want to update the prod DB manually, and I do not want anyone else to do that either. I see no reason we can't directly fall into a best practice process here. This simply means leaving 001 unaltered. Currently 001 does reflect the live DB state, if we run integration tests in travis we are simulating production. As soon as we merge this, we no longer get those benefits.

Reply:
> Wait, TripleA does.. As a moderator in lobby you can right click to ban by IP, right? It's not the most useful of features in its current form, but it is there to my knowledge.

@DanVanAtta Not anymore it seems, I just checked. You removed the controller classes yourself in fe55b9888317175398e698a70cef9a2f15f108a3 ^^
The RMI code still exists though, removing it would introduce incopatible changes anyways

Reply:
heh, indeed, hurray for commit comments!  :)
The backward incompatible is not desired just yet unless we are really ready to make that leap, good consideration @RoiEXLab . 
I would leave as is for now, we need to clean stuff up pretty big time and figure out an effective way to do moderator controls.. Looks like mac banning is not effective either (if the mac is coming from the client code and not network, then it's even more trivial to hack).